x/bsonx/bsoncore/value.go

```diff
@@ -208,63 +208,69 @@ func (v Value) Equal(v2 Value) bool {
 	return bytes.Equal(v.Data, v2.Data)
 }
 
+func idHex(id [12]byte) string {
+	var buf [24]byte
+	hex.Encode(buf[:], id[:])
+	return string(buf[:])
+}
+
 // String implements the fmt.String interface. This method will return values in extended JSON
 // format. If the value is not valid, this returns an empty string
 func (v Value) String() string {
 	switch v.Type {
-	case bsontype.Double:
+	case TypeDouble:
 		f64, ok := v.DoubleOK()
 		if !ok {
 			return ""
 		}
 		return fmt.Sprintf(`{"$numberDouble":"%s"}`, formatDouble(f64))
-	case bsontype.String:
+	case TypeString:
 		str, ok := v.StringValueOK()
 		if !ok {
 			return ""
 		}
 		return escapeString(str)
-	case bsontype.EmbeddedDocument:
+	case TypeEmbeddedDocument:
 		doc, ok := v.DocumentOK()
 		if !ok {
 			return ""
 		}
 		return doc.String()
-	case bsontype.Array:
+	case TypeArray:
 		arr, ok := v.ArrayOK()
 		if !ok {
 			return ""
 		}
 		return arr.String()
-	case bsontype.Binary:
+	case TypeBinary:
 		subtype, data, ok := v.BinaryOK()
 		if !ok {
 			return ""
 		}
 		return fmt.Sprintf(`{"$binary":{"base64":"%s","subType":"%02x"}}`, base64.StdEncoding.EncodeToString(data), subtype)
-	case bsontype.Undefined:
+	case TypeUndefined:
 		return `{"$undefined":true}`
-	case bsontype.ObjectID:
+	case TypeObjectID:
 		oid, ok := v.ObjectIDOK()
 		if !ok {
 			return ""
 		}
-		return fmt.Sprintf(`{"$oid":"%s"}`, oid.Hex())
-	case bsontype.Boolean:
+		return fmt.Sprintf(`{"$oid":"%s"}`, idHex(oid))
```

Comment:
The `v.ObjectIDOK()` at L254 returns `[12]byte` in this PR to avoid the import cycle. See the change at [L523](https://github.com/mongodb/mongo-go-driver/blob/2ba80e8a3fe34d8a8ed193ee251939acf06b088e/x/bsonx/bsoncore/value.go#L523).